spot-vaults/contracts/_strategies/SpotAppraiser.sol

```diff
@@ -0,0 +1,257 @@
+
+    /// @notice Contract constructor.
+    /// @param ampl Address of the AMPL token.
+    /// @param spot Address of the SPOT token.
+    /// @param usdOracle Address of the USD token market price oracle token.
+    /// @param amplOracle Address of the AMPL market price oracle token.
+    constructor(
+        IAMPL ampl,
+        IPerpetualTranche spot,
+        IChainlinkOracle usdOracle,
+        IChainlinkOracle amplOracle
+    ) Ownable() {
+        AMPL = ampl;
+        SPOT = spot;
+        AMPL_ORACLE = amplOracle;
+        USD_ORACLE = usdOracle;
+        UNIT_AMPL = 10 ** IERC20Metadata(address(AMPL)).decimals();
+
+        tolAMPLPriceDeviationPercs = Range({
+            lower: (ONE * 8) / 10, // 0.8
+            upper: (ONE * 7) / 4 // 1.75
+        });
+        minSPOTDR = (ONE * 8) / 10; // 0.8
+        minSeniorCDR = (ONE * 5) / 4; // 125%
+    }
+
+    //--------------------------------------------------------------------------
+    // Owner only methods
+
+    /// @notice Controls the tolerable AMPL price volatility outside which the pricing strategy is considered unreliable.
+    /// @param tolAMPLPriceDeviationPercs_ The lower and upper percentage deviation of AMPL price.
+    function updateAllowedPriceDeviationPercs(
+        Range memory tolAMPLPriceDeviationPercs_
+    ) external onlyOwner {
+        if (
+            tolAMPLPriceDeviationPercs_.lower > ONE ||
+            tolAMPLPriceDeviationPercs_.upper < ONE
+        ) {
+            revert InvalidPerc();
+        }
+        tolAMPLPriceDeviationPercs = tolAMPLPriceDeviationPercs_;
+    }
+
+    /// @notice Controls the minimum `deviationRatio` ratio of SPOT below which SPOT is considered unhealthy.
+    /// @param minSPOTDR_ The minimum SPOT `deviationRatio`.
+    function updateMinSPOTDR(uint256 minSPOTDR_) external onlyOwner {
+        minSPOTDR = minSPOTDR_;
+    }
+
+    /// @notice Controls the minimum CDR of SPOT's senior tranche below which SPOT is considered unhealthy.
+    /// @param minSeniorCDR_ The minimum senior tranche CDR.
+    function updateMinPerpCollateralCDR(uint256 minSeniorCDR_) external onlyOwner {
+        if (minSeniorCDR_ < ONE) {
+            revert InvalidSeniorCDRBound();
+        }
+        minSeniorCDR = minSeniorCDR_;
+    }
+
+    //--------------------------------------------------------------------------
+    // External methods
+
+    /// @return p The price of the usd token in dollars.
+    /// @return v True if the price is valid and can be used by downstream consumers.
+    function usdPrice() external view override returns (uint256, bool) {
+        (uint256 p, bool v) = _getCLOracleData(USD_ORACLE);
+        // If the market price of the USD coin fallen too much below 1$,
+        // it's an indication of some systemic issue with the USD token
+        // and thus its price should be considered unreliable.
+        return (ONE, (v && p > USD_LOWER_BOUND));
+    }
+
+    /// @return p The price of the spot token in dollars.
```

Comment:
To be ultra specific--this is the price in "dollars", not price in "dollar coins", right?